### PR TITLE
Handle errors accessing the amend state file

### LINKED ISSUE #9036: [NBug] Не удалось найти часть пути "E:\test_screenshots\.git\GitEx...
:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Error Details
```
System.IO.DirectoryNotFoundException: Не удалось найти часть пути "E:\test_screenshots\.git\GitExtensions.amend".
   в System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   в System.IO.File.InternalDelete(String path, Boolean checkHost)
   в System.IO.File.Delete(String path)
   в System.IO.Abstractions.FileWrapper.Delete(String path)
   в GitCommands.CommitMessageManager.set_AmendState(Boolean value)
   в GitUI.CommandsDialogs.FormCommit.OnFormClosed(FormClosedEventArgs e)
   в System.Windows.Forms.Form.CheckCloseDialog(Boolean closingOnly)
```

## Additional information
The repository has been deleted manually, but windows with commit was still opened. Trying to close the window

## Environment
- Git Extensions 3.5.0.11509
- Build 009d10c765a90aa00e043e40d5117256b793bc8c (Dirty)
- Git 2.30.0.windows.2
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 144dpi (150% scaling)

### PR DESCRIPTION
**Pull request by mstv**
Fixes #9036

## Proposed changes

- Handle errors accessing the amend state file the same way as for the commit message file

## Screenshots 

N/A

## Test methodology 

- manual
- slightly adapted existing NUnit tests

## Test environment(s) 

- Git Extensions 33.33.33
- Build 28d662737516d4f7d41611eb2a84ba18e240d58f
- Git 2.27.0.windows.1 (recommended: 2.30.0 or later)
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/CommitMessageManager.cs [hunk 1]
```diff
@@ -227,14 +201,44 @@ internal static string FormatCommitMessage(string commitMessage, bool usingCommi
...
+                if (!_fileSystem.File.Exists(filePath))
+                {
+                    return string.Empty;
+                }
+
+                return _fileSystem.File.ReadAllText(filePath, encoding ?? Encoding.Default);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(null, string.Format(CannotAccessFile, ex.Message, filePath), errorTitle,
```
Comment (RussKie) [inline GitCommands/CommitMessageManager.cs:219]: Let it bubble up to the global handler?

Comment (mstv) [reply GitCommands/CommitMessageManager.cs:219]: No, I agree to what you wrote "in this specific scenario we need to ignore the exception".
Just let the user know that something went wrong but I see no need to cancel the other operations in FormCommit.

Comment (mstv) [reply GitCommands/CommitMessageManager.cs:219]: (If these file accesses would fail often, a retry should be offered here.)

Comment (RussKie) [reply GitCommands/CommitMessageManager.cs:219]: I think we should leave a comment here explaining why we're swallowing the exception and not sending to the global handler.

**Review by RussKie — APPROVED** [review]

**Comment by mstv** [comment]
msftbot merge in 1 day

**Comment by mstv** [comment]
@msftbot merge in 1 day

**Pull request merged**